### PR TITLE
#808: Update flavor standard-EXASOL-7.1.0 to Ubuntu 20.04

### LINKED ISSUE #808: Update flavor standard-EXASOL-7.1.0 to Ubuntu 20.04
# Background

- Currently, the flavor standard-EXASOL-7.1.0  uses Ubuntu 18.04 which will be soon end of life
- This update will likely produce larger issues, because standard-EXASOL-7.1.0 uses currently python3.7, however this is not officially supported by Ubuntu 20.04 and will also be arriving at end of life at end of June 2023
  - This means we need to update this container to python 3.8 which will be a breaking change
  - The update to python 3.8 might also cause an update of the Python packages, because of incompatibilities 

# Acceptance Criteria

- https://github.com/exasol/script-languages-release/issues/836
- Update to Ubuntu 20.04 and python 3.8
- Update to the newest pyexasol version to support connections to v8

### LINKED ISSUE #824: Update scipy for standard 7.* flavor to fix CVE-2023-29824
For the moment, we ignore CVE-2023-29824 for the standard 7.* flavor, because the fixed scipy 1.8.0 is not available for python 3.7 and we first need to upgrade these flavors.

- [x] https://github.com/exasol/script-languages-release/issues/808
- [x] https://github.com/exasol/script-languages-release/issues/807

### LINKED ISSUE #836: ModuleNotFoundError: No module named 'numpy.testing.nosetester'
I was trying to use UDF for simple ML model on small dataset.  I encountered the following error.

```
SQL Error [22002]: VM error: F-UDF-CL-LIB-1125: F-UDF-CL-SL-PYTHON-1000: F-UDF-CL-SL-PYTHON-1017: ExaUDFError: F-UDF-CL-SL-PYTHON-1122: Exception while parsing UDF 
ML_CREATE_MODEL_CHURN:2 <module>
  File "/usr/lib/python3/dist-packages/sklearn/model_selection/__init__.py", line 19, in <module>
    from ._validation import cross_val_score
  File "/usr/lib/python3/dist-packages/sklearn/model_selection/_validation.py", line 29, in <module>
    from ..metrics.scorer import check_scoring, _check_multimetric_scoring
  File "/usr/lib/python3/dist-packages/sklearn/metrics/__init__.py", line 7, in <module>
    from .ranking import auc
  File "/usr/lib/python3/dist-packages/sklearn/metrics/ranking.py", line 25, in <module>
    from scipy.stats import rankdata
  File "/usr/lib/python3/dist-packages/scipy/stats/__init__.py", line 348, in <module>
    from .stats import *
  File "/usr/lib/python3/dist-packages/scipy/stats/stats.py", line 177, in <module>
    from . import distributions
  File "/usr/lib/python3/dist-packages/scipy/stats/distributions.py", line 13, in <module>
    from . import _continuous_distns
  File "/usr/lib/python3/dist-packages/scipy/stats/_continuous_distns.py", line 15, in <module>
    from scipy._lib._numpy_compat import broadcast_to
  File "/usr/lib/python3/dist-packages/scipy/_lib/_numpy_compat.py", line 10, in <module>
    from numpy.testing.nosetester import import_nose
ModuleNotFoundError: No module named 'numpy.testing.nosetester'
 (Session: 1774555151640559616)
```

Exasol Version: Docker DB 7.1.9

UDF: 

```
--/
CREATE OR REPLACE PYTHON3 SET SCRIPT ML_CREATE_MODEL_CHURN(...) EMITS ("Accuracy" FLOAT) AS 
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score


def run(ctx):
    # Preprocess the data (convert categorical to numerical)
    df = ctx.get_dataframe(num_rows='all', start_col=0)#pd.DataFrame(data_)
    data_encoded = pd.get_dummies(df, columns=["State","Phone","Int'l Plan","VMail Plan"])
    target_encoded = pd.get_dummies(df['Churn?'])  # 'Churn?' is the categorical target variable
    # Split data into features and target
    X = data_encoded.drop('Churn?', axis=1)
    y = target_encoded  # Use the encoded target variable
    # Split into training and testing sets
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

    # Create a Random Forest classifier model
    rf_model = RandomForestClassifier(n_estimators=100, random_state=42)

    # Train the model
    rf_model.fit(X_train, y_train)

    # Predict on the test set
    predictions = rf_model.predict(X_test)

    # Evaluate the model
    accuracy = accuracy_score(y_test, predictions)
    #print(f"Accuracy: {accuracy}")
    # Model needs TO be saved IN BucketFS FROM here 
    # Another UDF picks the model frm BucketFS AND Scores the TABLE
    # OPEN issue, we need TO fix the SLC Sklearn version

    ctx.emit(accuracy)
/
```

### PR DESCRIPTION
**Pull request by tkilias**
Fixes #808 
Fixes #824
Fixes #836

### All Submissions:

* [x] Check if your pull request goes to the correct bash branch (develop or master)?
* [x] Is the title of the Pull Request correct?
* [x] Is the title of the corresponding issue correct?
* [x] Have you updated the changelog?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../../pulls) for the same update/change?  
* [x] Are you mentioning the issue which this PullRequest fixes ("Fixes...")

### When integrating to Develop branch:

1. [ ] Remember to merge with "Squash and Merge"

### When integrating to Main branch:

1. [ ] Remember to merge with "Merge"

### CODE REVIEW

File: flavors/standard-EXASOL-7.1.0/flavor_base/base_test_deps/packages/apt_get_packages [hunk 1]
```diff
@@ -1,7 +1,7 @@
-gdb|8.1.1-0ubuntu1
-valgrind|1:3.13.0-2ubuntu2.3
-gdbserver|8.1.1-0ubuntu1
-binutils|2.30-21ubuntu1~18.04.9
-patchelf|0.9-1
+gdb|9.2-0ubuntu1~20.04.1                                                                                                 
+valgrind|1:3.15.0-1ubuntu9.1                                                                                        
+gdbserver|9.2-0ubuntu1~20.04.1                                                                                           
+binutils|2.34-6ubuntu1.6                                                                                         
+patchelf|0.10-2build1
```
Comment (kaklakariada) [inline flavors/standard-EXASOL-7.1.0/flavor_base/base_test_deps/packages/apt_get_packages:5]:
```suggestion
gdb|9.2-0ubuntu1~20.04.1
valgrind|1:3.15.0-1ubuntu9.1
gdbserver|9.2-0ubuntu1~20.04.1
binutils|2.34-6ubuntu1.6
patchelf|0.10-2build1
```

trailing whitespace

File: flavors/standard-EXASOL-7.1.0/flavor_customization/Dockerfile [hunk 2]
```diff
@@ -20,7 +20,7 @@ COPY flavor_customization/packages/apt_get_packages /build_info/packages/flavor_
 RUN /scripts/install_scripts/install_via_apt.pl --file /build_info/packages/flavor_customization/apt_get_packages --with-versions
 
 COPY flavor_customization/packages/python3_pip_packages /build_info/packages/flavor_customization
-RUN /scripts/install_scripts/install_via_pip.pl --file /build_info/packages/flavor_customization/python3_pip_packages --python-binary python3.7 --with-versions --allow-no-version  --use-deprecated-legacy-resolver --ignore-installed
+RUN /scripts/install_scripts/install_via_pip.pl --file /build_info/packages/flavor_customization/python3_pip_packages --python-binary python3.8 --with-versions --allow-no-version  --use-deprecated-legacy-resolver --ignore-installed
```
Comment (kaklakariada) [inline flavors/standard-EXASOL-7.1.0/flavor_customization/Dockerfile:23]:
```suggestion
RUN /scripts/install_scripts/install_via_pip.pl --file /build_info/packages/flavor_customization/python3_pip_packages --python-binary python3.8 --with-versions --allow-no-version
```
in another place you also removed this

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: flavors/standard-EXASOL-7.1.0/flavor_customization/Dockerfile (hunk 2 above)
Comment (tkilias) [reply flavors/standard-EXASOL-7.1.0/flavor_customization/Dockerfile:23]: Good catch, I actually forgot this

File: flavors/standard-EXASOL-7.1.0/flavor_base/base_test_deps/packages/apt_get_packages (hunk 1 above)
Comment (tkilias) [reply flavors/standard-EXASOL-7.1.0/flavor_base/base_test_deps/packages/apt_get_packages:5]: Ah, that comes from copying out of tmux, I hoped I got all

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**